docs/howto/add_link.md

```diff
@@ -0,0 +1,336 @@
+
+```ruby
+class UserSerializer < ActiveModel::Serializer
+  attributes :id, :name
+  has_many :posts do
+    link(:self) do
+      user_posts_path(object.id) # Rails url helper automatically available
+    end 
+  end
+end
+
+user = User.new(id: 1, name: "Kelly", posts: [])
+resource = ActiveModelSerializers::SerializableResource.new(user, adapter: :json_api)
+resource.to_json
+```
+
+Note that `user_posts_path` method is one of Rails' url helpers, which are automatically available within the `link` block if you are using Rails.
+
+The previous snippet would result in:
+```json
+{
+  "data": {
+    "id": "1",
+    "type": "users",
+    "attributes": {
+      "name": "Kelly"
+    },
+    "relationships": {
+      "posts": {
+        "data": [],
+        "links": {
+          "self": "users/1/posts"
+        }
+      }
+    }
+  }
+}
+```
+
+If you only want the `links` member to appear you can set `include_data false` within the relationship block:
+
+```ruby
+class UserSerializer < ActiveModel::Serializer
+  attributes :name
+  has_many :posts do
+    link(:self) do
+      include_data false
+      user_posts_path(object.id) # Rails url helper automatically available
+    end 
+  end
+end
+
+user = User.new(id: 1, name: "Kelly", posts: [ Post.new(id: 1) ])
+resource = ActiveModelSerializers::SerializableResource.new(user, adapter: :json_api)
+resource.to_json
+```
+
+This would result in:
+
+```json
+{
+  "data": {
+    "id": "1",
+    "type": "users",
+    "attributes": {
+      "name": "Kelly"
+    },
+    "relationships": {
+      "posts": {
+        "links": {
+          "self": "users/1/posts"
+        }
+      }
+    }
+  }
+}
+```
+
+## JSON adapter
+
+The `json` is not able to handle links defined by the `link` class method. However, you can either use the `meta`  member or define a `links` attribute on the serializer.
+
+### Via `meta`
+
+```ruby
+user = User.new(id: 1, name: "John")
+meta = {
+  links: {
+    self: "/users/#{user.id}",
+    posts: "/posts?user_id=#{user.id}"
+  }
+}
+resource = ActiveModelSerializer::SerializableResource.new(
+  user,
+  adapter: :json,
+  meta: meta
+)
+resource.to_json
+```
+
+This would result in:
+
+```json
+{
+  "meta": {
+    "links": {
+      "self": "/users/1",
+      "posts": "/posts?user_id=1"
+    }
+  },
+  "user": {
+    "id": "1",
+    "name": "Example User",
+  }
+}
+```
+
+When using Rails, you can add links by providing the `meta` option to `render` method:
+
+```ruby
+class UserController < ActionController::Base
+  def show
+    user = User.new(id: 1, name: "John")
+     meta = {
+      links: {
+        self: "/users/#{user.id}",
+        posts: "/posts?user_id=#{user.id}"
+      }
+    }
+    render json: user, adapter: :json, meta: meta
+  end
+end
+```
+
+### Via an attribute
+
+```ruby
+class UserSerializer < ActiveModel::Serializer
+  attributes :id, :name, :links
+  attribute :links do
```

Comment:
well, that goes back to our design goals of making all serializers/adapters support links